File: ClientApp/src/app/interfaces/Product/Response/productResponse.ts
import { ByBaseResponse } from "../../byBaseResponse";
import { CategoriesListResponse } from "./categoriesListResponse";
import { OptionsResponse } from "./optionsResponse";
import { SuppliersListResponse } from "./suppliersListResponse";
import { WarrantiesListResponse } from "./warrantiesListResponse";

export interface ProductResponse extends ByBaseResponse {
    id: string;
    code: string;
    isService: boolean;
    option: OptionsResponse;
    category: CategoriesListResponse;
    warranty: WarrantiesListResponse | null;
    supplier: SuppliersListResponse | null;
    options: OptionsResponse[];
}

File: ClientApp/src/app/interfaces/Product/Response/ProductResponse.cs
namespace Contracts.Model.Product.Response;

public class ProductResponse : ByBaseResponse
{
	public Guid Id { get; set; }
	public string Code { get; set; }
	public bool IsService { get; set; }
	public int CategoryId { get; set; }
	public Guid? WarrantyId { get; set; }
	public Guid? SupplierId { get; set; }
	public ICollection<OptionsResponse> Options { get; set; }
}